client_options.go

```diff
@@ -3,68 +3,105 @@ package walletclient
-	return func(c *WalletClient) {
-		if c != nil {
-			c.xPubString = xPubString
-		}
+// XPriv sets the xPrivString field of a WalletClient
+type XPriv struct {
+	XPrivString *string
+}
+
+func (w *XPriv) Configure(c *WalletClient) {
+	var err error
+	if c.xPriv, err = bitcoin.GenerateHDKeyFromString(*w.XPrivString); err != nil {
+		c.xPriv = nil
 	}
 }
 
-// WithAccessKey will set the access key on the client
-func WithAccessKey(accessKeyString string) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.accessKeyString = accessKeyString
-		}
+// XPub sets the xPubString on the client
+type XPub struct {
+	XPubString *string
+}
+
+func (w *XPub) Configure(c *WalletClient) {
+	var err error
+	if c.xPub, err = bitcoin.GetHDKeyFromExtendedPublicKey(*w.XPubString); err != nil {
+		w.XPubString = nil
 	}
+
 }
 
-// WithHTTP will overwrite the default client with a custom client
-func WithHTTP(serverURL string) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.transportOptions = append(c.transportOptions, transports.WithHTTP(serverURL))
-		}
+// AccessKey sets the accessKeyString on the client
+type AccessKey struct {
+	AccessKeyString *string
+}
+
+func (w *AccessKey) Configure(c *WalletClient) {
+	var err error
+	if c.accessKey, err = w.initializeAccessKey(); err != nil {
+		c.accessKey = nil
 	}
 }
 
-// WithHTTPClient will overwrite the default client with a custom client
-func WithHTTPClient(serverURL string, httpClient *http.Client) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.transportOptions = append(c.transportOptions, transports.WithHTTPClient(serverURL, httpClient))
-		}
+// AdminKey sets the admin key for creating new xpubs
+type AdminKey struct {
+	AdminKeyString *string
+}
+
+func (w *AdminKey) Configure(c *WalletClient) {
+	var err error
+	c.adminXPriv, err = bitcoin.GenerateHDKeyFromString(*w.AdminKeyString)
+	if err != nil {
+		c.adminXPriv = nil
 	}
 }
 
-// WithAdminKey will set the admin key for admin requests
-func WithAdminKey(adminKey string) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.transportOptions = append(c.transportOptions, transports.WithAdminKey(adminKey))
-		}
+// HTTP sets the URL and HTTP client of a WalletClient
+type HTTP struct {
+	ServerURL  *string
+	HTTPClient *http.Client
+}
+
+func (w *HTTP) Configure(c *WalletClient) {
+	c.server = *w.ServerURL
+	c.httpClient = w.HTTPClient
+	if w.HTTPClient != nil {
+		c.httpClient = w.HTTPClient
+	} else {
+		c.httpClient = http.DefaultClient
 	}
 }
 
-// WithSignRequest will set whether to sign all requests
-func WithSignRequest(signRequest bool) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.transportOptions = append(c.transportOptions, transports.WithSignRequest(signRequest))
+// SignRequest configures whether to sign HTTP requests
+type SignRequest struct {
+	Sign *bool
+}
+
+func (w *SignRequest) Configure(c *WalletClient) {
+	c.signRequest = *w.Sign
+}
```

Comment:
About my suggestion about not storing pointers in configurator.
Consider my suggestion:
```golang
type SignRequest struct {
	Sign bool
}

func (w *SignRequest) Configure(c *WalletClient) {
	c.signRequest = w.Sign
}
```